pkg/scheduler/preemption/preemption.go

```diff
@@ -323,12 +301,41 @@ func fairPreemptions(wl *workload.Info, assignment flavorassigner.Assignment, sn
 				}
 				// Might need to pick a different CQ due to changing values.
 				break
+			} else {
+				retryCandidates = append(retryCandidates, candCQ.workloads[i])
```

Comment:
Shouldn't we only retry these that are `belowThreshold`?

Reply:
if it's above threshold, it would not come to the `else`.

Reply:
Hm, the first condition is `belowThreshold || newNominatedShareValue <= newCandShareVal`, so if above threshold it depends on the result of `newNominatedShareValue <= newCandShareVal`, which could be either way. Unless I'm missing something.

Reply:
True.
However, it doesn't matter. The workload is still a candidate, under the eye of fair sharing.

Reply:
The scenario I'm thinking about is that it is above the threshold, and `newNominatedShareValue <= newCandShareVal ` is false, then it is added to `retryCandidates`. 

In that case it is considered in rule B, where we no longer check the threshold. So, in the end a workload above threshold could be preempted if selected in rule B. Wondering if this is expected?